mkt/extensions/validation.py

```diff
@@ -70,10 +79,13 @@ def __init__(self, file_obj=None):
         self.file_obj = file_obj
         self.zipfile = None
 
-    def error(self, error_key):
+    def error(self, error_key, error_parameter=None):
+        message = self.errors[error_key]
+        if error_parameter is not None:
+            message = self.errors[error_key] % error_parameter
         raise ParseError(detail={
             'key': error_key,
-            'message': self.errors[error_key],
+            'message': message,
         })
```

Comment:
Not sure I made the intent of this totally clear. If using what I suggested above results in a response body of:

    {
        'key': 'ICON_INCORRECT_DIMENSIONS',
        'message': 'The icon file `256.png` dimensions are different from the size',
        'params': [
            '256.png'
        ]
    }

Then we can use a combination of `key` + `params` to construct a far more helpful error message.